payrole_frontend/src/components/layout/Sidebar.tsx

```
@@ -0,0 +1,227 @@
+    // Check permissions based on user role
+    if (item.id === 'employees' && !canManageEmployees()) return false;
+    if (item.id === 'payments' && !canProcessPayments()) return false;
+    if (item.id === 'treasury' && !canManageTreasury()) return false;
+    if (item.id === 'admin' && !canAccessAdmin()) return false;
+    
+    return true;
+  });
+
+  const renderMenuItem = (item: MenuItem, isChild = false) => {
+    const hasChildren = item.children && item.children.length > 0;
+    const [isExpanded, setIsExpanded] = React.useState(false);
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)